src/google/protobuf/parse_context.h

```diff
@@ -208,7 +208,7 @@ class PROTOBUF_EXPORT EpsCopyInputStream {
   bool DoneWithCheck(const char** ptr, int d) {
     GOOGLE_DCHECK(*ptr);
     if (PROTOBUF_PREDICT_TRUE(*ptr < limit_end_)) return false;
-    int overrun = *ptr - buffer_end_;
+    size_t overrun = *ptr - buffer_end_;
```

Comment:
I believe this has to be a signed type because `overrun` can be negative. Maybe `std::ptrdiff_t` would be the right type.

Reply:
Consider: `auto overrun = *ptr - buffer_end_;`

Reply:
finally, I made a new commit with
`    int overrun = static_cast<int>(*ptr - buffer_end_);`

overrun is used for compare with limit_ (which is int) and as first argument of DoneFallback (which expect int).